ts/main.ts

```diff
@@ -194,49 +263,187 @@ async function renderAllCardsInitial(): Promise<void> {
     cardsArray.push(renderCard(spellInfo.name, spellInfo.level, spellInfo.url));
   }
 
-  sortCardsName();
+  sortCards('name');
 }
 
 renderAllCardsInitial();
 
-function sortCardsName(): void {
-  cardsArray.sort((a, b) => {
-    const firstName = a.getAttribute('data-name') as string;
-    const secondName = b.getAttribute('data-name') as string;
-    return firstName.localeCompare(secondName);
-  });
-  cardsArray.forEach((element) => {
-    $spellsListCardsDiv.appendChild(element);
-  });
+// an array.sort callback function to sort cards by their data-name attribute
+function nameSort(a: HTMLDivElement, b: HTMLDivElement): number {
+  const firstName = a.getAttribute('data-name') as string;
+  const secondName = b.getAttribute('data-name') as string;
+  return firstName.localeCompare(secondName);
 }
 
-function sortCardsLevel(): void {
-  cardsArray.sort(
-    (a, b) =>
-      Number(a.getAttribute('data-level')) -
-      Number(b.getAttribute('data-level')),
+// an array.sort callback function to sort cards by their data-level attribute
+function numberSort(a: HTMLDivElement, b: HTMLDivElement): number {
+  return (
+    Number(a.getAttribute('data-level')) - Number(b.getAttribute('data-level'))
   );
+}
+
+// a function that will sort the cards in both the all cards array and the
+// filtered cards array
+function sortCards(criteria: string): void {
+  if (criteria === 'name') {
+    cardsArray.sort(nameSort);
+    filteredArray.sort(nameSort);
+  } else if (criteria === 'level') {
+    cardsArray.sort(numberSort);
+    filteredArray.sort(numberSort);
+  }
+
   cardsArray.forEach((element) => {
     $spellsListCardsDiv.appendChild(element);
   });
+  filteredArray.forEach((element) => {
+    $spellsListFilteredCardsDiv.appendChild(element);
+  });
 }
 
+// event listener for when the sort order is changed
 $spellsListSortDropdown.addEventListener('input', () => {
+  cardSort.sort = $spellsListSortDropdown.value;
+
   // clear cards
   while ($spellsListCardsDiv.childNodes.length > 0) {
     if (!$spellsListCardsDiv.firstElementChild) break;
     $spellsListCardsDiv.removeChild($spellsListCardsDiv.firstElementChild);
   }
 
   // re-render cards based on sort value
-  if ($spellsListSortDropdown.value === 'level') {
-    sortCardsLevel();
+  sortCards(cardSort.sort);
```

Comment:
Was this from the last feature? Would we be able to do this without deleting and recreating all of the cards?